app/models/compute_resources/foreman/model/libvirt.rb

```diff
@@ -132,7 +132,11 @@ def template(id)
       template
     end
 
+    # The `create_vm_ctx` flag indicates whether `new_vm` is being called with the intent
+    # to actually create a VM. If the flag is not set, the call may be for purposes like
+    # validation (e.g., triggered by the UI) without provisioning a VM.
```

Comment:
`create_vm_ctx` is still not clear IMO. How about calling it `creation_mode`?
```suggestion
    # The `creation_mode` flag indicates whether the new_vm method should actually provision a virtual machine.
    # If set to false, the method may be used for validation, preview, or UI-related logic without performing any real provisioning actions.
```